inspirehep/modules/workflows/views/callback.py

```diff
@@ -54,41 +54,57 @@ def _get_base_url():
     return base_url
 
 
-@blueprint.route('/workflows/continue', methods=['POST'])
-def continue_workflow_callback():
-    """Handle callback to continue a workflow.
+def _continue_workflow(workflow_id, recid, result=None):
+    """Small wrapper to continue a workflow.
 
-    Expects the request data to contain a object ID in the
-    nonce field.
+    Will prepare the needed data from the record id and the result data if
+    passed.
+
+    :return: True if succeeded, False if the specified workflow id does not
+        exist.
     """
-    request_data = request.get_json()
-    id_object = request_data.get("nonce", "")
-
-    if id_object:
-        callback_results = request_data.get("results", {})
-        workflow_object = workflow_object_class.get(id_object)
-        if workflow_object:
-            results = request_data.get("results", [])
-            for result in results:
-                status = result.get('success', False)
-                if status:
-                    recid = result.get('recid')
-                    base_url = _get_base_url()
-                    workflow_object.extra_data['url'] = join(
-                        base_url,
-                        'record',
-                        str(recid)
-                    )
-                    workflow_object.extra_data['recid'] = recid
-            # Will add the results to the engine extra_data column.
-            workflow_object.save()
-            db.session.commit()
-            workflow_object.continue_workflow(
-                delayed=True,
-                callback_results=callback_results
-            )
-            return jsonify({"result": "success"})
-    return jsonify({"result": "failed"})
+    base_url = _get_base_url()
+    workflow_object = workflow_object_class.get(workflow_id)
+    if not workflow_object:
+        current_app.logger.error(
+            'No workflow object with the id %s could be found.',
+            workflow_id,
+        )
+        return False
+
+    workflow_object.extra_data['url'] = join(
+        base_url,
+        'record',
+        str(recid)
+    )
+    workflow_object.extra_data['recid'] = recid
```

Comment:
So in all cases that you call `_continue_workflow` you will have a recid available?